src/core/ext/transport/chttp2/transport/incoming_metadata.cc

```diff
@@ -38,7 +38,8 @@ grpc_error* grpc_chttp2_incoming_metadata_buffer_add(
     storage = static_cast<grpc_linked_mdelem*>(
         buffer->arena->Alloc(sizeof(grpc_linked_mdelem)));
   }
-  return grpc_metadata_batch_add_tail(&buffer->batch, storage, elem);
+  storage->md = elem;
```

Comment:
any special reason for this? Are we rolling out all functions like these?

Reply:
Putting aside my confusion that both methods exist - I like keeping small operations like setting a qword to be on the near side of a compilation unit barrier, giving the compiler the freedom to move things around if needed during optimization. Not saying it makes a huge difference in this case, though.